ergotree-interpreter/src/eval/scoll.rs

```diff
@@ -122,6 +124,28 @@ pub(crate) static ZIP_EVAL_FN: EvalFn = |_env, _ctx, obj, args| {
     }
 };
 
+pub(crate) static INDICES_EVAL_FN: EvalFn = |_env, _ctx, obj, _args| {
+    let normalized_input_vals: Vec<Value> = match obj {
+        Value::Coll(coll) => Ok(coll.as_vec()),
+        _ => Err(EvalError::UnexpectedValue(format!(
+            "expected obj to be Value::Coll, got: {0:?}",
+            obj
+        ))),
+    }?;
+    let indices = normalized_input_vals
+        .into_iter()
+        .enumerate()
+        .flat_map(|(i, _)| i32::try_from(i))
```

Comment:
I assume out of i32 bounds `i` here got silently dropped. Although it might never happen (huge `Coll`) let's "bubble up" `EvalError`.  `collect::<Result<Vec<i32>, _>>()` can be made out of `Vec<Result<i32, _>>`.